test/unit/spriteManipulation.js

```diff
@@ -251,6 +251,55 @@ test('Sprite dance changes will throw with invalid parameters', async t => {
   t.end();
 });
 
+test('Sprite move sorting works reliably', async t => {
+  const moveNames = [
+    {
+      name: 'rest',
+      rest: true,
+    },
+    {
+      name: 'fullLength1',
+    },
+    {
+      name: 'fullLength2',
+    },
+    {
+      name: 'shortBurst1',
+      shortBurst: true,
+    },
+    {
+      name: 'shortBurst2',
+      shortBurst: true,
+    },
+  ];
+
+  const P5 = require('../../src/loadP5');
+
+  P5.prototype.loadJSON = function (_url, callback) {
```

Comment:
Why not `sinon.stub(P5.prototype, 'loadJSON').callsFake(...)` here? Or if this is just to get initialization working, should we do this inside the `createDanceAPI` helper?

Reply:
`sinon`: while I added while experimenting with it, we don't currently have it in this project. We could add it for this case if you think it helps.

Inside `createDanceAPI`: I'm not sure we're going to want to override these is any consistent way throughout all of the test cases. These specific overrides are just good enough to test the sorting, but nothing else. Maybe we can come up with some great overrides that return fixtures and use them everywhere. Or maybe we'll switch to karma/phantom and use sinon fakeserver.

Reply:
I added sinon two days ago in https://github.com/code-dot-org/dance-party/pull/161 for testing the measure counter. (Fake timers are really useful for some of the behaviors in this repo).